kernels/portable/cpu/op_grid_sampler_2d.cpp

```diff
@@ -0,0 +1,257 @@
+
+          // Calculate output index
+          const int64_t out_idx = n * out_batch_stride +
+              c * out_channel_stride + h * out_height_stride + w;
+
+          // Apply interpolation method
+          if (interpolation_mode == 0) { // Bilinear
+            // Calculate corner indices and weights
+            auto [ix_nw, lambda_x] =
+                compute_source_index_and_weight(x, inp_W, align_corners);
+            auto [iy_nw, lambda_y] =
+                compute_source_index_and_weight(y, inp_H, align_corners);
+
+            // Calculate bilinear weights
+            float w_nw = (1.0f - lambda_x) * (1.0f - lambda_y);
+            float w_ne = lambda_x * (1.0f - lambda_y);
+            float w_sw = (1.0f - lambda_x) * lambda_y;
+            float w_se = lambda_x * lambda_y;
+
+            // Get corner pixel values with boundary checking
+            T nw = get_pixel_value(
+                input_data, n, c, iy_nw, ix_nw, inp_H, inp_W, padding_mode);
+            T ne = get_pixel_value(
+                input_data, n, c, iy_nw, ix_nw + 1, inp_H, inp_W, padding_mode);
+            T sw = get_pixel_value(
+                input_data, n, c, iy_nw + 1, ix_nw, inp_H, inp_W, padding_mode);
+            T se = get_pixel_value(
+                input_data,
+                n,
+                c,
+                iy_nw + 1,
+                ix_nw + 1,
+                inp_H,
+                inp_W,
+                padding_mode);
+
+            // Perform bilinear interpolation (weighted sum)
+            out_data[out_idx] =
+                static_cast<T>(nw * w_nw + ne * w_ne + sw * w_sw + se * w_se);
+          } else if (interpolation_mode == 1) { // Nearest
+            // Convert to pixel space and round to nearest pixel
+            float ix = unnormalize_coord(x, inp_W, align_corners);
+            float iy = unnormalize_coord(y, inp_H, align_corners);
+
+            int64_t ix_nearest = static_cast<int64_t>(std::round(ix));
+            int64_t iy_nearest = static_cast<int64_t>(std::round(iy));
+
+            // Get nearest pixel value
+            out_data[out_idx] = get_pixel_value(
+                input_data,
+                n,
+                c,
+                iy_nearest,
+                ix_nearest,
+                inp_H,
+                inp_W,
+                padding_mode);
+          }
+        }
+      }
+    }
+  }
+}
+
+// Main grid_sampler_2d function that validates inputs and dispatches to
+// implementation
+Tensor& grid_sampler_2d_out(
+    KernelRuntimeContext& ctx,
+    const Tensor& input,
+    const Tensor& grid,
+    int64_t interpolation_mode,
+    int64_t padding_mode,
+    bool align_corners,
+    Tensor& out) {
+  const int64_t N = input.size(0);
+  const int64_t C = input.size(1);
+  const int64_t out_H = grid.size(1);
+  const int64_t out_W = grid.size(2);
+
+  // Check for 4D input and grid
+  ET_KERNEL_CHECK(ctx, (input.dim() == 4), InvalidArgument, out);
```

Comment:
@DenisVieriu97 do you mind refactoring all of the InvalidArgument checks into a helper function dedicated to check the arguments, which returns True only if all checks pass, and logs when a check doesn't pass. Here is an [example](https://github.com/pytorch/executorch/blob/main/kernels/portable/cpu/util/upsample_util.cpp#L15-L39)
and then you would be calling
`ET_KERNEL_CHECK(ctx, check_grid_sampler_2d_args(...) == 4), InvalidArgument, out);`